modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4029,6 +4029,7 @@ protected void updateDataTypeWithEnumForMap(CodegenProperty property) {
 
         if (baseItem != null) {
             // set both datatype and datetypeWithEnum as only the inner type is enum
+            property.dataType = property.dataType.replace(", " + baseItem.baseType, ", " + toEnumName(baseItem));
```

Comment:
Thanks for the PR. I think `dataType` should not be updated to do anything with enum.

That's why we've datatypeWithEnum.

The reason is that some languages do not support Enum (and some implementation itself doesn't yet support enum even though the language does).